src/annotator/guest.ts

```diff
@@ -598,6 +598,7 @@ export class Guest implements Annotator, Destroyable {
    *   frame about the removal of an anchor.
    */
   detach(tag: string, notify = true) {
+    this._clusterToolbar?.scheduleClusterUpdates();
```

Comment:
When adding annotations, `detach()` is called at the start of the anchoring process, but there are async steps that happen before the new highlights are added, and those could potentially take longer than whatever timeout the re-processing of highlights use internally. Also this would break if someone tried to optimize `anchor()` in future by not calling `detach` if there were no existing anchors.

What you could do instead is move this logic into `this._updateAnchors`, since that is already called after detaching highlights and adding new ones, in order to trigger an update of the bucket bar.

Reply:
Indeed when testing in some PDFs (eg. http://localhost:3000/pdf/gatsby) I sometimes don't see the increased saturation for overlapping highlights immediately after the document loads. Moving the update call to the end of `this._updateAnchors` resolves the issue.

Reply:
Related to this, the effect is lost after a nested highlight is hovered in the sidebar and then-unhovered. After doing that, a sequence of overlapping highlights have a uniform color.

Reply:
> Related to this, the effect is lost after a nested highlight is hovered in the sidebar and then-unhovered. After doing that, a sequence of overlapping highlights have a uniform color.

Actually, I believe a separate and potentially-complex issue. When annotations that correspond to SVG highlights are hovered, the SVG highlights are actually re-ordered in the documents such that the highlighted SVG(s) are the last child of the parent (i.e. "come last"). This is a whole other tangent...thanks for indirectly alerting me to it!